### PR TITLE
Twin sequences

### PR DESCRIPTION
**Pull request by ben-mulhern**

### CODE REVIEW

File: src/prime.ts [hunk 1]
```diff
@@ -25,8 +25,7 @@ function* generator(
   prev: Array<number> = []
 ): Generator<number> {
   yield current;
-  const largestPrimeSoFar = prev[prev.length - 1];
-  const nextPrime = findNextPrime(nextCandidate(largestPrimeSoFar), prev);
-  prev.push(nextPrime);
+  prev.push(current);
+  const nextPrime = findNextPrime(nextCandidate(current), prev);
```
Comment (ben-mulhern) [inline src/prime.ts:29]: This is just a tidy up - slightly nicer way of doing it.

File: src/twin.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+// Generator
+function* generator(
+  a: Seq.Indexed<number>,
+  b: Seq.Indexed<number>,
+  counter: number = 0
+): Generator<number> {
+  // I'm annoyed that this doesn't work!
+  //   yield counter % 2 === 0 ? a.get(counter / 2) : b.get((counter - 1) / 2);
+  yield counter % 2 === 0 ? a.first() : b.first();
```
Comment (ben-mulhern) [inline src/twin.ts:23]: This was infuriating and vexed me for ages. Basically these Seqs-wrapping-Generators are not immutable!
If you do take(5) on one of them - the first time you call it you get the first five elements - fine. But call take(5) on the Seq a second time and you get the next five elements.

So I am using first() here as kind of a next() equivalent.

File: src/__tests__/twin.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Seq } from "immutable";
```
Comment (mikeyhogarth) [inline src/__tests__/twin.spec.ts:1]: Why you including `Seq` here?

File: src/twin.ts (hunk 2 above)
Comment (mikeyhogarth) [reply src/twin.ts:23]: Oh shit! Will `first` always give you the first elements? So much for immutable js!

File: src/index.ts [hunk 4]
```diff
@@ -5,3 +5,4 @@ export { default as Geometric } from "./geometric";
 export { default as Power } from "./power";
 export { default as Prime } from "./prime";
 export { default as Triangle } from "./triangle";
+export { default as Twin } from "./twin";
```
Comment (mikeyhogarth) [inline src/index.ts:8]: FYI: I added this earlier - this is basically the main entry point for people importing this library , so if you add new sequences then you need to add a line to this file to make them available outside.

**Pull request merged**